synapse/logging/opentracing.py

```diff
@@ -793,48 +793,41 @@ def _tag_args_inner(*args, **kwargs):
     return _tag_args_inner
 
 
-def trace_servlet(servlet_name, extract_context=False):
-    """Decorator which traces a serlet. It starts a span with some servlet specific
-    tags such as the servlet_name and request information
+@contextlib.contextmanager
+def trace_servlet(request: "SynapseRequest", extract_context: bool = False):
+    """Context manager which traces a servlet. It starts a span with some
+    servlet specific tags such as the servlet_name and request information.
```

Comment:
I don't think this is accurate. Particularly, it does nothing with the servlet_name. Generally I don't think it traces a servlet (and could do with a rename): perhaps "traces a request"?

Reply:
also: aiui a "context manager" is a thing you can call `with` on... so this is a method that *returns* a context manager, rather than being one itself.